phpcs-sniffs/PluginCheck/Sniffs/CodeAnalysis/DiscouragedFunctionsSniff.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * DiscouragedFunctionsSniff
+ *
+ * Based on code from {@link https://github.com/WordPress/WordPress-Coding-Standards}
+ * which is licensed under {@link https://opensource.org/licenses/MIT}.
+ *
+ * @package PluginCheck
+ */
+
+namespace PluginCheckCS\PluginCheck\Sniffs\CodeAnalysis;
+
+use PHPCSUtils\Utils\MessageHelper;
+use PHPCSUtils\Utils\PassedParameters;
+use WordPressCS\WordPress\AbstractFunctionRestrictionsSniff;
+
+/**
+ * Detect discouraged functions.
+ *
+ * @link https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/
+ *
+ * @since 1.6.0
+ */
+final class DiscouragedFunctionsSniff extends AbstractFunctionRestrictionsSniff {
+
+	/**
+	 * Groups of functions to discourage.
+	 *
+	 * Example: groups => array(
+	 *  'lambda' => array(
+	 *      'type'      => 'error' | 'warning',
+	 *      'message'   => 'Use anonymous functions instead please!',
+	 *      'functions' => array( 'file_get_contents', 'create_function' ),
+	 *  )
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'load_plugin_textdomain' => array(
+				'type'      => 'warning',
+				'message'   => 'Using %s() for loading the plugin translations is not needed for WordPress.org directory since WordPress 4.6.',
```

Comment:
Can this sniff take into account the plugin's `Requires at least` value accordingly? If it still supports older versions, then the call can still be there.

Reply:
PR has been updated.